joern-cli/frontends/jimple2cpg/src/main/scala/io/joern/jimple2cpg/Main.scala

```diff
@@ -13,21 +14,22 @@ final case class Config(inputPaths: Set[String] = Set.empty, outputPath: String
   override def withOutputPath(x: String): Config = copy(outputPath = x)
 }
 
-/** Entry point for command line CPG creator
-  */
-object Main extends App {
+private object Frontend {
+
+  implicit val defaultConfig: Config = Config()
 
-  private val frontendSpecificOptions = {
+  val cmdLineParser: OParser[Unit, Config] = {
     val builder = OParser.builder[Config]
     import builder.programName
     OParser.sequence(programName("jimple2cpg"))
   }
 
-  X2Cpg.parseCommandLine(args, frontendSpecificOptions, Config()) match {
-    case Some(config) =>
-      new Jimple2Cpg().run(config)
-    case None =>
-      System.exit(1)
+  def run(config: Config, jimple2Cpg: Jimple2Cpg): Unit = {
```

Comment:
haven't really thought this through, but looks like we could make `run` an abstract function on `X2CpgMain`, and then not pass it as a constructor argument